hack/gen-test-flavors.sh

```diff
@@ -22,7 +22,8 @@ set -o pipefail
 root=$(dirname "${BASH_SOURCE[0]}")/..
 kustomize="${root}/hack/tools/bin/kustomize"
 test_dir_path="test/e2e/data/infrastructure-aws"
-test_dir="${root}/${test_dir_path}"
+sub_dir_path=$0
+test_dir="${root}/${test_dir_path}/${sub_dir_path}"
```

Comment:
@richardcase fixed based on [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3754/files/a0f0d668908cf06302b9f719d2c9c16aca0aa9c9#r1000313512) comment.